src/generative.jl

```diff
@@ -1,59 +1,119 @@
-# generative model interface functions
-
 """
-    generate_s{S,A}(p::Union{POMDP{S,A},MDP{S,A}}, s::S, a::A, rng::AbstractRNG)
+    gen(...)
 
-Return the next state given current state `s` and action taken `a`.
-"""
-function generate_s end
+Sample from generative model of a POMDP or MDP.
 
-"""
-    generate_o{S,A,O}(p::POMDP{S,A,O}, s::S, a::A, sp::S, rng::AbstractRNG)
+There are 3 versions:
+- For problem-writers, the most convenient version to implement is gen(m::Union{MDP,POMDP}, s, a, rng::AbstractRNG), which returns a `NamedTuple`.
+- Solvers and simulators should use the version with a `DBNOut` argument.
+- Defining behavior for and sampling from individual nodes of the dynamic Bayesian network can be accomplished using the version with a `DBNVar` argument.
```

Comment:
I think it is a bit confusing as to whether you want people to implement specialized versions of `gen(v::DBNVar, ...)` (as stated here) or `gen(v::DBNOut,...)` (as stated in `generative.md`). I understand that both is possible but we should clarify in the documentation, when to do what.

Reply:
fixed generative.md